server/src/database.rs

```diff
@@ -275,28 +330,28 @@ impl Iterator for TransactionsIterator {
             return None;
         }
 
-        let read_transaction = |mut content: &mut io::Cursor<_>| -> Result<_> {
+        let read_transaction = |content: &mut T, position: &mut u64| -> Result<_> {
             let mut sender = [0u8; 20];
             let mut hash = [0u8; 32];
             let rlp_len = content.read_u32::<LittleEndian>()? as usize;
+            *position += 4;
             let mut rlp = Vec::with_capacity(rlp_len);
             rlp.resize(rlp_len, 0);
             content.read_exact(&mut sender)?;
+            *position += 20;
             content.read_exact(&mut hash)?;
+            *position += 32;
             content.read_exact(&mut rlp)?;
+            *position += rlp_len as u64;
             Ok(Transaction::new(sender.into(), hash.into(), rlp))
```

Comment:
this isthis looks less readable and more error prone then it used to. is there any reason to change that?